contrib/ci/check-source.py

```diff
@@ -110,6 +112,25 @@ def _test_lines_gerror(self, lines: List[str]) -> None:
                         self.add_failure("uses g_set_error() without using FWUPD_ERROR")
                         break
 
+    def _test_lines_function_length(self, lines: List[str]) -> None:
+        # these files are allowed to have a huge main()
+        if os.path.basename(self._current_fn) in ["fu-tool.c", "fu-util.c"]:
```

Comment:
I was thinking another way to avoid it can be `//nocheck` **just** on the function. Then it can still run the rest of those files.

Reply:
i actually tried that -- then we skip the other tests unintentionally. We could have `/* nOcHeCk */` or some other variant that doesn't have the `nocheck`string...

Reply:
maybe all the `nocheck`'s need qualification, e.g. `nocheck:function-length` or something like that?

Reply:
Yeah that's good!

Reply:
https://github.com/fwupd/fwupd/pull/7674

Reply:
Done, and rebased. Good idea!